integrations/opensearch/src/haystack_integrations/document_stores/opensearch/document_store.py

```diff
@@ -113,13 +116,39 @@ def client(self) -> OpenSearch:
                 "`settings` values will be ignored.",
                 self._index,
             )
-        else:
+        elif self._create_index:
             # Create the index if it doesn't exist
             body = {"mappings": self._mappings, "settings": self._settings}
-
             self._client.indices.create(index=self._index, body=body)  # type:ignore
         return self._client
 
+    def create_index(
```

Comment:
Do we need this one to be public?

Reply:
Use-case for us is the following:
- whenever a pipeline is created, we want to create the index, but only in this situation.
- if the pipeline should actually be down but is still running (e.g. because embedding batches can take a couple of seconds), but the index has already been deleted async, the index shouldn't be created again automatically

To support this:
- we can set `create_index=False` init param for `OpenSearchDocumentStore`
- call `create_index()` on the document store of the pipeline whenever we load the pipeline

I'm not opinionated here if it has to be public. But I can imagine this as valuable in general whenever you don't want the index to be created automatically. For this I would see it as public, but just let me know if I should change it.